### PR TITLE
DDP 98 - Add workflow to push docker image to dockerhub

### PR DESCRIPTION
**Pull request by d-jeph**
# Add workflow to push docker image to dockerhub

- This PR adds a Github action workflow(`docker.yml`) to push the prefect proxy image to the [tech4dev](https://hub.docker.com/u/tech4dev) dockerhub registry whenever a new release is published
- The image is tagged with the release version number

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Login to docker registry
+        uses: docker/login-action@v3
+        with:
+          username: ${{secrets.DOCKERHUB_USERNAME}}
+          password: ${{secrets.DOCKERHUB_PASSWORD}}
+
+      - name: Build and push docker image to registry
+        uses: docker/build-push-action@v5
+        with:
+          context: .
```
Comment (yvonnegitau) [inline .github/workflows/docker.yml:33]: Could you add multi-platform `platforms: linux/amd64,linux/arm64`

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  push_image_to_registry:
+    name: Push Docker image to docker hub
+    runs-on: ubuntu-latest
+    steps:
```
Comment (yvonnegitau) [inline .github/workflows/docker.yml:14]: Do we need to setup buildx? ``` - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v3 ```

**Review by yvonnegitau — APPROVED** [review]
Looks good. Left some comments. Have you tested

File: .github/workflows/docker.yml (hunk 1 above)
Comment (d-jeph) [reply .github/workflows/docker.yml:39]: Added this

File: .github/workflows/docker.yml (hunk 2 above)
Comment (d-jeph) [reply .github/workflows/docker.yml:14]: included this and caching as well

**Pull request merged**